scripts/rotating_camera/controllers/rotating_camera/rotating_camera.c

```diff
@@ -0,0 +1,145 @@
+//  o height_step: Height achieved during a complete revolution.
+//  o angle_end: Make an arc of a circle. Set to 6.28 for complete revolution.
+
+static void revolving_view(double *robot_position, int rotation_direction, int vertical_direction, int n_turn, float radius, float height_step, double angle_end) {
+
+  int k = 0;
+  double angle = 0.0;
+  double angle_step = 0.01;
+  double new_position[3] = {0.0, 0.0, 0.0};
+  double orientation[4] = {0, 1, 0, 0}; // Front view
+
+  // Get Viewpoint node and get the fields "position" and "orientation"
+  viewpoint = wb_supervisor_field_get_mf_node(children, 1); // Search ViewPoint node
+  viewpoint_position = wb_supervisor_node_get_field(viewpoint, "position");
+  viewpoint_orientation = wb_supervisor_node_get_field(viewpoint, "orientation");
+
+  while (k < n_turn){ // Start at angle 0 and end at angle n_turn*2*PI.
+
+    if (fabs(angle_end - ANGLE_REV)>TOL) {
+      if (angle >= angle_end) {
+        break;
+      }
+    }
+    if (angle >= ANGLE_REV) { // had make a revolution
+      angle = 0.0;
+      k += 1;
+    }
+
+    // Set new position to ViewPoint node
+    new_position[0] = robot_position[0] + radius * sin(angle);
+    if (vertical_direction == UP)
+      new_position[1] += height_step/(6.28/angle_step);
+    else
+      new_position[1] -= height_step/(6.28/angle_step);
+    new_position[2] = robot_position[2] + radius * cos(angle);
+    wb_supervisor_field_set_sf_vec3f(viewpoint_position, new_position);
+
+    // Set new orientation to ViewPoint node
+    orientation[3] = angle;
+    wb_supervisor_field_set_sf_rotation(viewpoint_orientation, orientation);
+
+    // Update the angle value
+    if (rotation_direction == CW)
+      angle += angle_step;
+    else // CounterClockWise
+      angle -= angle_step;
+
+    wb_robot_step(wb_robot_get_basic_time_step()); // wait for 32 milliseconds
+  }
+}
+
+
+
+int main(int argc, char *argv[]) {
+
+  const double *temp;
+  double robot_position[3] = {0.0, 0.0, 0.0};
+
+  // Init and get the scene tree (=root)
+  wb_robot_init();
+
+  root = wb_supervisor_node_get_root();
+  children = wb_supervisor_node_get_field(root, "children");
+
+  // Count and display all nodes in the world
+  int n = wb_supervisor_field_get_count(children);
+  printf("This world contains %d nodes:\n", n);
+  for (int i = 0; i < n; i++) {
+    nodes = wb_supervisor_field_get_mf_node(children, i);
+    printf("-> %s\n", wb_supervisor_node_get_type_name(nodes));
+  }
+  printf("\n");
+
+  // In Robot node, get the field "translation"
+  robot = wb_supervisor_field_get_mf_node(children, 5); // Search Robot node
+  robot_translation = wb_supervisor_node_get_field(robot, "translation");
+  temp = wb_supervisor_field_get_sf_vec3f(robot_translation);
+  robot_position[0] = temp[0];
+  robot_position[1] = temp[1];
+  robot_position[2] = temp[2];
+
+  printf("Start in 1 second (simulation time)...\n");
+  wb_robot_step(1000); /* wait for 2 seconds */
```

Comment:
This should be a multiple of basic time step
```suggestion
  wb_robot_step(1024);  // wait for about 1 second
```